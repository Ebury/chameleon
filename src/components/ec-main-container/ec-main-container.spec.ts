
import { type ComponentMountingOptions, mount } from '@vue/test-utils';

import EcMainContainer from './ec-main-container.vue';
import type { MainContainerProps } from './types';

describe('EcMainContainer', () => {
  function mountEcMainContainer(props?: Partial<MainContainerProps>, mountOpts: ComponentMountingOptions<MainContainerProps> = {}) {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    return mount(EcMainContainer as any, {
      props,
      ...mountOpts,
    });
  }

  it('should render empty if no props were given', () => {
    const wrapper = mountEcMainContainer();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render both titles when they are given in props', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mountEcMainContainer(props);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not display the titleIntro when title is not given', () => {
    const props = {
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mountEcMainContainer(props);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render without title intro when the titleIntro prop is not given', () => {
    const props = {
      title: 'Trade Finance',
    };
    const wrapper = mountEcMainContainer(props);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use given slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mountEcMainContainer(props, {
      slots: {
        default: '<p>Random text</p>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use the given named cta slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mountEcMainContainer(props, {
      slots: {
        cta: '<button class="ec-btn ec-btn--rounded ec-btn--primary ec-btn--md ec-btn--full-width">Test cta</button>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should use the given named breadcrumbs slot', () => {
    const props = {
      title: 'Trade Finance',
      titleIntro: 'Here you will be able to keep track of all your requests to Ebury and your credit line.',
    };
    const wrapper = mountEcMainContainer(props, {
      slots: {
        breadcrumbs: '<a href="#">Breadcrumbs</a>',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
