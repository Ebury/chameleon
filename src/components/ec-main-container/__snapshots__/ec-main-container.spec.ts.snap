// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EcMainContainer should display breadcrumbs even when there are no props 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <div
    class="tw-flex tw-justify-between tw-items-center tw-mb-16"
  >
    <div
      data-test="ec-main-container__breadcrumbs"
    >
      
      <a
        href="#"
      >
        Breadcrumbs
      </a>
      
    </div>
    <!--v-if-->
  </div>
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;

exports[`EcMainContainer should render both titles when they are given in props 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <!--v-if-->
  <div
    class="tw-flex tw-justify-between tw-items-center tw-flex-wrap-reverse tw--mt-16"
  >
    <h1
      class="ec-main-container__title"
      data-test="ec-main-container__title"
    >
      Trade Finance
    </h1>
    <!--v-if-->
  </div>
  <p
    class="ec-main-container__title-intro"
    data-test="ec-main-container__title-intro"
  >
    Here you will be able to keep track of all your requests to Ebury and your credit line.
  </p>
  
  
</div>
`;

exports[`EcMainContainer should render empty if no props or slots were given 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <!--v-if-->
  <!--v-if-->
  <!--v-if-->
  
  
</div>
`;

exports[`EcMainContainer should render without title intro when the titleIntro prop is not given 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <!--v-if-->
  <div
    class="tw-flex tw-justify-between tw-items-center tw-flex-wrap-reverse tw--mt-16"
  >
    <h1
      class="ec-main-container__title"
      data-test="ec-main-container__title"
    >
      Trade Finance
    </h1>
    <!--v-if-->
  </div>
  <!--v-if-->
  
  
</div>
`;

exports[`EcMainContainer should use given slot 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <!--v-if-->
  <div
    class="tw-flex tw-justify-between tw-items-center tw-flex-wrap-reverse tw--mt-16"
  >
    <h1
      class="ec-main-container__title"
      data-test="ec-main-container__title"
    >
      Trade Finance
    </h1>
    <!--v-if-->
  </div>
  <p
    class="ec-main-container__title-intro"
    data-test="ec-main-container__title-intro"
  >
    Here you will be able to keep track of all your requests to Ebury and your credit line.
  </p>
  
  <p>
    Random text
  </p>
  
</div>
`;

exports[`EcMainContainer should use the given named breadcrumbs slot 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <div
    class="tw-flex tw-justify-between tw-items-center tw-mb-16"
  >
    <div
      data-test="ec-main-container__breadcrumbs"
    >
      
      <a
        href="#"
      >
        Breadcrumbs
      </a>
      
    </div>
    <!--v-if-->
  </div>
  <div
    class="tw-flex tw-justify-between tw-items-center tw-flex-wrap-reverse tw--mt-16"
  >
    <h1
      class="ec-main-container__title"
      data-test="ec-main-container__title"
    >
      Trade Finance
    </h1>
    <!--v-if-->
  </div>
  <p
    class="ec-main-container__title-intro"
    data-test="ec-main-container__title-intro"
  >
    Here you will be able to keep track of all your requests to Ebury and your credit line.
  </p>
  
  
</div>
`;

exports[`EcMainContainer should use the given named cta slot 1`] = `
<div
  class="ec-main-container"
  data-test="ec-main-container"
>
  <!--v-if-->
  <div
    class="tw-flex tw-justify-between tw-items-center tw-flex-wrap-reverse tw--mt-16"
  >
    <h1
      class="ec-main-container__title"
      data-test="ec-main-container__title"
    >
      Trade Finance
    </h1>
    <div
      class="tw-ml-auto tw-mt-16"
      data-test="ec-main-container__cta"
    >
      
      <button
        class="ec-btn ec-btn--rounded ec-btn--primary ec-btn--md ec-btn--full-width"
      >
        Test cta
      </button>
      
    </div>
  </div>
  <p
    class="ec-main-container__title-intro"
    data-test="ec-main-container__title-intro"
  >
    Here you will be able to keep track of all your requests to Ebury and your credit line.
  </p>
  
  
</div>
`;
